aptos-move/block-executor/src/executor.rs

```diff
@@ -236,25 +236,26 @@ where
             ));
         }
 
-        let record_succeeded = last_input_output.record(idx_to_execute, read_set, result);
-        InternalExecuteResult {
-            record_succeeded,
-            updates_outside,
+        if !last_input_output.record(idx_to_execute, read_set, result) {
+            return Err(PanicOr::Or(ModulePathReadWrite));
```

Comment:
nit: it might be cleaner for record to `return` this error? Otherwise a bit ambiguous where this is coming from